src/java.base/share/classes/java/time/LocalDate.java

```diff
@@ -2147,28 +2155,58 @@ public int hashCode() {
      */
     @Override
     public String toString() {
-        int yearValue = year;
-        int monthValue = month;
-        int dayValue = day;
-        int absYear = Math.abs(yearValue);
-        StringBuilder buf = new StringBuilder(10);
-        if (absYear < 1000) {
-            if (yearValue < 0) {
-                buf.append(yearValue - 10000).deleteCharAt(1);
-            } else {
-                buf.append(yearValue + 10000).deleteCharAt(0);
+        byte[] buf = new byte[yearSize(year) + 6];
+        getChars(buf, 0);
+
+        try {
+            return jla.newStringNoRepl(buf, StandardCharsets.ISO_8859_1);
+        } catch (CharacterCodingException cce) {
+            throw new AssertionError(cce);
+        }
+    }
+
+    static int yearSize(int year) {
+        if (Math.abs(year) < 1000) {
+            return year < 0 ? 5 : 4;
+        }
+        return jla.stringSize(year) + (year > 9999 ? 1 : 0);
+    }
+
+    int getChars(byte[] buf, int off) {
+        int year = this.year;
+        int yearSize = yearSize(year);
+        int yearAbs = Math.abs(year);
+        if (yearAbs < 1000) {
+            if (year < 0) {
+                buf[off] = '-';
```

Comment:
`buf[off++] = '-';`

Reply:
this place doesn't need off++

Reply:
It was a suggestion, implicitly paired with removing ` + (year < 0 ? 1 : 0)` from line 2188.

Reply:
The reason for not using off++ is that it can be executed out of order, which may result in better performance.

Reply:
I don't think that would outweigh the extra branch and the increased code size. I'd opt for simplicity.

Reply:
I've simplified LocalDate::getChars based on your suggestion